File: solutions/cpp/pangram/1/pangram.cpp
#include <set>

#include "pangram.h"

namespace pangram {
	bool is_pangram(std::string input) {
		for (char c = 'a'; c <= 'z'; c++) {
			if (input.find(c) == std::string::npos && input.find(c - 32) == std::string::npos)
				return false;
		}

		return true;
	}
}  // namespace pangram

File: solutions/go/pangram/1/pangram.go
// Package pangram implements a function to determine if a sentence is a pangram.
package pangram

import "unicode"

// IsPangram determines if a sentence is a pangram.
func IsPangram(input string) bool {
	lettersFound := make([]bool, 26)

	for _, char := range input {
		if unicode.IsLetter(unicode.ToLower(char)) {
			lettersFound[unicode.ToLower(char)-'a'] = true
		}
	}

	for _, found := range lettersFound {
		if !found {
			return false
		}
	}

	return true
}
